selfdrive/car/toyota/carstate.py

```diff
@@ -94,6 +94,14 @@ def update(self, cp, cp_cam):
       ret.cruiseState.available = cp.vl["PCM_CRUISE_2"]["MAIN_ON"] != 0
       ret.cruiseState.speed = cp.vl["PCM_CRUISE_2"]["SET_SPEED"] * CV.KPH_TO_MS
 
+    is_metric = cp.vl["BODY_CONTROL_STATE_2"]["UNITS"] in (1, 2)
+    conversion_factor = CV.KPH_TO_MS if is_metric else CV.MPH_TO_MS
+    if is_metric:
+      ret.vEgoCluster = cp.vl["BODY_CONTROL_STATE_2"]["UI_SPEED"] * CV.KPH_TO_MS
+    else:
+      ret.vEgoCluster = cp.vl["BODY_CONTROL_STATE_2"]["UI_SPEED"] * CV.KPH_TO_MS * 1.05
```

Comment:
It seems like `UI_SPEED` is not the actual UI speed shown in the dash. For one, it updates visually at a different rate and lags behind dash. Secondly, it cannot show 1 mph/kph (skips from 2 kph to 0 kph) while the dash can. And finally, `UI_SET_SPEED` is actually in the car's unit preference, however `UI_SPEED` stays in metric no matter what (the comment in the DBC was copied over without verifying).

I found similar signals in 0x3cb and 0x361 (CLUTCH), however they didn't exist on all cars and were also always metric. The dash/cluster must being doing this conversion internally, and then slowing down the rate to 2 Hz itself, without publishing the signal anywhere, so looks like we need to do the same here.